api/scholarqa/rag/multi_step_qa_pipeline.py

```diff
@@ -101,130 +97,6 @@ def make_prompt(query: str, paper_paper_quotes_dict: Dict[str, str]) -> str:
                                       )
         return json.loads(response.content), response
 
-    def get_quote_citations(self, retrieval_df: pd.DataFrame, per_paper_summaries: Dict[str, str],
-                            plan_json: Dict[str, List[int]]) -> Dict[str, List[str]]:
-        """
-        Map the quotes extracted in step 1 `per_paper_summaries` to their actual full length versions
-        in retrieval_df and make a map of quote ref string --> [inline citations].
-
-        i) First parse the plan json to get the required paper indices to be sent for the final generation in step 3.
-        ii) For those papers, find the quotes from per_paper_summaries and passages from retrieval_df
-        iii) Try to match the paper quotes as substrings to the corresponding paper passages to get the inline citations along with
-        offsets.
-        iv) First try matching the raw string (high precision), if unsuccessful, try matching only the alphabet characters (high recall).
-        v) Once a match is found, use its offsets in the passage and iterate over the inline citations to find any that occur within the snippet.
-        vi) In case of a raw string match, replace any citation mentions (if possible), with the paper id of the corresponding inline citation to be linked later.
-        """
```

Comment:
Moved the code to the scholarQA wrapper class, more appropriate home for the code